packages/iov-bns-governance/src/governor.spec.ts

```diff
@@ -0,0 +1,243 @@
+    });
+  });
+
+  describe("getProposals", () => {
+    it("can get an empty list of proposals", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity(HdPaths.iov(100));
+      const governor = new Governor(options);
+
+      const proposals = await governor.getProposals();
+      expect(proposals.length).toEqual(0);
+
+      options.connection.disconnect();
+    });
+  });
+
+  describe("createProposalTx", () => {
+    it("throws an error for unsupported proposal types", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      await governor
+        .createProposalTx({
+          type: ProposalType.AddCommitteeMember,
+          description: "Change something",
+          startTime: new ReadonlyDate(1562164525898),
+          electionRuleId: 5,
+          committee: 8 as CommitteeId,
+          address: "abcd" as Address,
+        })
+        .then(
+          () => fail("Expected promise to be rejected"),
+          err => expect(err.message).toMatch(/proposal type not yet supported/i),
+        );
+
+      options.connection.disconnect();
+    });
+
+    it("can create a CreateProposal transaction for AmendProtocol", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      const tx = await governor.createProposalTx({
+        type: ProposalType.AmendProtocol,
+        text: "Switch to Proof-of-Work",
+        description: "Proposal to change consensus algorithm to POW",
+        startTime: new ReadonlyDate(1562164525898),
+        electionRuleId: 1,
+      });
+      expect(tx).toEqual({
+        kind: "bns/create_proposal",
+        creator: options.identity,
+        title: "Amend protocol",
+        action: {
+          resolution: "Switch to Proof-of-Work",
+        },
+        description: "Proposal to change consensus algorithm to POW",
+        electionRuleId: 1,
+        startTime: 1562164525,
+        author: bnsCodec.identityToAddress(options.identity),
+        fee: {
+          tokens: {
+            quantity: "10000000",
+            fractionalDigits: 9,
+            tokenTicker: "CASH" as TokenTicker,
+          },
+        },
+      });
+
+      options.connection.disconnect();
+    });
+
+    it("can create a CreateProposal transaction with a custom title", async () => {
+      pendingWithoutBnsd();
+      const options = await getConnectionAndIdentity();
+      const governor = new Governor(options);
+
+      const tx = await governor.createProposalTx({
+        type: ProposalType.AmendProtocol,
+        text: "Switch to Proof-of-Work",
+        title: "Custom title",
+        description: "Proposal to change consensus algorithm to POW",
+        startTime: new ReadonlyDate(1562164525898),
+        electionRuleId: 1,
+      });
+      expect((tx as any).title).toEqual("Custom title");
```

Comment:
`tx` should be of type `CreateProposalTx & WithCreator`, which would avoid the cast